corehq/apps/accounting/forms.py

```diff
@@ -1854,27 +1861,23 @@ def trigger_customer_invoice(self):
 
     @staticmethod
     def clean_previous_invoices(invoice_start, invoice_end, account):
-        invoices = CustomerInvoice.objects.filter(
+        prev_invoices = CustomerInvoice.objects.filter(
             date_start__lte=invoice_end,
             date_end__gte=invoice_start,
             account=account
         )
-        prev_invoices = []
-        for invoice in invoices:
-            if invoice.account == account:
-                prev_invoices.append(invoice)
         if prev_invoices:
             from corehq.apps.accounting.views import CustomerInvoiceSummaryView
             raise InvoiceError(
                 "Invoices exist that were already generated with this same "
                 "criteria. You must manually suppress these invoices: "
                 "{invoice_list}".format(
-                    num_invoices=len(prev_invoices),
+                    num_invoices=prev_invoices.count(),
                     invoice_list=', '.join(
                         ['<a href="{edit_url}">{name}</a>'.format(
                             edit_url=reverse(CustomerInvoiceSummaryView.urlname, args=(x.id,)),
                             name=x.invoice_number
-                        ) for x in prev_invoices]
+                        ) for x in prev_invoices.all()]
```

Comment:
Curious, is this is a stylistic, functional, or performance change? Not popping out to me immediately what the `.all()` here does differently.

Reply:
I actually can't remember why I changed this in the first place. Thank you for pointing that out!